### PR TITLE
Add new story page type

### PR DESCRIPTION
**Pull request by contolini**
Adds a new "Story Page" content type with a sidebar-less centered column and images that bleed into the left and right gutters. It's use case is prose-heavy landing pages designed to tell a story.

To let editors choose when they want either RichTextBlock images or FullWidthText images to bleed into the gutters, a 'bleed' format option has been added to both image fields. Selecting this option for an image that's not on a story page will have no affect (it'll be treated as a standard full width image).

See context at https://github.local/CFPB/el-camino/issues/406

## Notes

- It'd be ideal to only show the "bleed" image format when editing a story page but I don't think that's possible given how the [formats are defined](https://docs.wagtail.io/en/stable/advanced_topics/customisation/page_editing_interface.html#rich-text-image-formats).
- Open to any styling recommendations. I'm using a [content modifier class](https://github.com/cfpb/consumerfinance.gov/compare/story-page?expand=1#diff-388c7690c0d544528764b456f7f7b7b76aa7758049ff55d062338dcf11c32083R22) to scope the CSS. Would a template level class (e.g. `.t-story`) be more appropriate? Or something else altogether? The `.bleed-image` class is vaguely named because it has to apply to both content-image molecules and rich text field images.

## Additions

- Story page type

## Changes

- "Bleed" image format option to both rich text images and full width text images.

## How to test this PR

1. I made a demo page here: https://cfgov-pr-6460.DEMO-PROXY/lorem-ipsum-dolor-sit-amet/
1. Log in with admin/admin to view the editing experience

or

1. Create a Story page.
1. Add a few full width text blocks.
1. Add images and rich text fields with images and they should have a new "bleed" image format.
1. Preview the page and the appropriate images should hang into the left/right gutters.

## Screenshots

| full-width text image selector | rich text image selector | preview |
|--------|--------|---------|
| editor   | editor   | preview    |

## Checklist

- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Future todos are captured in comments and/or tickets
- [x] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

#### Browser testing

Visually tested in the following supported browsers:
- [x] Firefox
- [x] Chrome
- [x] Safari
- [x] Edge 18 (the last Edge prior to it switching to Chromium)
- [x] Internet Explorer 11 and 8 (via emulation in 11's dev tools)
- [x] Safari on iOS
- [x] Chrome on Android

#### Accessibility

- [x] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [x] Screen reader friendly
- [x] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [x] Is useable without CSS
- [x] Is useable without JS
- [x] Does not introduce new lint warnings
- [x] Flexible from small to large screens

### CODE REVIEW

File: cfgov/v1/jinja2/v1/story_page.html
Comment (anselmbradford) [inline cfgov/v1/jinja2/v1/story_page.html:12]: Could this be `block__flush-bottom` instead?

Comment (contolini) [reply cfgov/v1/jinja2/v1/story_page.html:12]: Let me try it. I directly copied the HTML from [campaign-page.html](https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/v1/jinja2/v1/campaign_page.html) which is the only other page (aside from the homepage) that uses a jumbo hero.

Comment (contolini) [reply cfgov/v1/jinja2/v1/story_page.html:12]: Hmmm yeah the utility class is used because `block__flush-bottom` removes the bottom border which we don't want.

screenshot

File: cfgov/unprocessed/css/enhancements/layout.less
Comment (anselmbradford) [inline cfgov/unprocessed/css/enhancements/layout.less:336]: We should probably settle on `bleed-image` or `image-bleed` (see `content__center-image-bleed` above)

Comment (anselmbradford) [reply cfgov/unprocessed/css/enhancements/layout.less:336]: Do we need `img` here? Maybe just have it `.bleed-image` (if we switched to `<picture>` or something down the road.

File: cfgov/v1/jinja2/v1/story_page.html
Comment (anselmbradford) [inline cfgov/v1/jinja2/v1/story_page.html:21]: Should be `{% block content_modifiers -%}`

File: cfgov/v1/jinja2/v1/story_page.html
Comment (anselmbradford) [inline cfgov/v1/jinja2/v1/story_page.html:22]: Technically this is not correct BEM. This is creating:
```
main.content
  div.content_wrapper
    div.content_main.content__center-image-bleed
    …
```

Buuuuut the modifier `content__center-image-bleed` should be modifying `content`, not `content_main`. So the structure should be:

```
main.content.content__center-image-bleed
  div.content_wrapper
    div.content_main
    …
```

See earlier comment.

File: cfgov/unprocessed/css/enhancements/layout.less
Comment (anselmbradford) [inline cfgov/unprocessed/css/enhancements/layout.less:328]: This would be `& .content_main > section {` if `content__center-image-bleed` is moved up to `content`.

Comment (anselmbradford) [reply cfgov/unprocessed/css/enhancements/layout.less:328]: We should probably give `section` a name [in the template](https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/jinja2/v1/_layouts/content-base-main-only.html#L7), like `content_main-inner` or something. Maaaaaybe something for another PR. Your call.

File: cfgov/unprocessed/css/enhancements/layout.less
Comment (anselmbradford) [inline cfgov/unprocessed/css/enhancements/layout.less:337]: Is this used?

File: cfgov/v1/jinja2/v1/story_page.html
Comment (anselmbradford) [inline cfgov/v1/jinja2/v1/story_page.html:5]: There's a use-case for `features` being added to this template?

Screen Shot 2021-05-27 at 1 49 58 PM

File: cfgov/unprocessed/css/enhancements/layout.less
Comment (contolini) [reply cfgov/unprocessed/css/enhancements/layout.less:337]: Yeah L342 and L343. The constant multiplication and division was visually overwhelming so I moved the basic gutter calcs to these variables.

File: cfgov/v1/jinja2/v1/story_page.html
Comment (contolini) [reply cfgov/v1/jinja2/v1/story_page.html:5]: The jumbo hero is designed to have optional features tucked underneath it. See [SSSU](https://www.consumerfinance.gov/start-small-save-up/) or the homepage. We won't be using it for our new inequity page but future story pages might need it.

File: cfgov/unprocessed/css/enhancements/layout.less
Comment (anselmbradford) [reply cfgov/unprocessed/css/enhancements/layout.less:337]: Oh I see. I had done a find search and didn't find it in the escaped string below. All good!

**Review by schbetsy — APPROVED** [review]

File: cfgov/unprocessed/css/enhancements/layout.less
Comment (contolini) [reply cfgov/unprocessed/css/enhancements/layout.less:328]: > We should probably give `section` a name [in the template](https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/jinja2/v1/_layouts/content-base-main-only.html#L7), like `content_main-inner` or something. Maaaaaybe something for another PR. Your call.

Sounds good to me. Seems harmless and helpful.

File: cfgov/unprocessed/css/enhancements/layout.less
Comment (contolini) [reply cfgov/unprocessed/css/enhancements/layout.less]: Changed everything to `.image-bleed` in https://github.com/cfpb/consumerfinance.gov/commit/e9f1d87096db0a1e546dadc7b3f2dc17c0bd8d2f 👍

File: cfgov/v1/jinja2/v1/story_page.html
Comment (contolini) [reply cfgov/v1/jinja2/v1/story_page.html]: Fixed in https://github.com/cfpb/consumerfinance.gov/commit/e9f1d87096db0a1e546dadc7b3f2dc17c0bd8d2f

File: cfgov/v1/jinja2/v1/story_page.html
Comment (contolini) [reply cfgov/v1/jinja2/v1/story_page.html:22]: Fixed in https://github.com/cfpb/consumerfinance.gov/commit/e9f1d87096db0a1e546dadc7b3f2dc17c0bd8d2f

File: cfgov/unprocessed/css/enhancements/layout.less
Comment (contolini) [reply cfgov/unprocessed/css/enhancements/layout.less]: Added the new class in  https://github.com/cfpb/consumerfinance.gov/commit/e9f1d87096db0a1e546dadc7b3f2dc17c0bd8d2f

File: cfgov/v1/image_formats.py
Comment (Scotchester) [inline cfgov/v1/image_formats.py:9]: Let's set this to the max width of a bleeding image so that it doesn't request originally-uploaded images that are unnecessarily large.

```suggestion
        'width-1170'
```

File: cfgov/v1/models/story_page.py
Comment (Scotchester) [inline cfgov/v1/models/story_page.py:12]: Did you consider just importing `CampaignHeader` instead of redefining it here as `StoryHeader`? I may be missing something, but seems like that would be a little more DRY.

File: cfgov/v1/models/story_page.py
Comment (Scotchester) [inline cfgov/v1/models/story_page.py:35]: This is not important and not wrong as-is, but just as something you might be interested to know, `StoryHeader` (`CampaignHeader`, originally) was defined as its own class so that its `Meta` `block_counts` could be set. When that `Meta` is not needed for a StreamField, as in the case of the `content` StreamField here, we usually just define the StreamField items within `content = StreamField(…, blank=True)`.

File: cfgov/v1/image_formats.py
Comment (Scotchester) [inline cfgov/v1/image_formats.py:4]: I have a small concern about adding the ability to add images in this format inside rich text fields. Most notably, it requires that alt text be added for each inserted image, which is a minor accessibility concern if the image is decorative (i.e., a screen reader reading text about the image wouldn't add value for the user). Aside from that, we have more control over the markup output when using our custom ContentImage block.

All that said, I know it can be annoying to have to break up long stretches of text with separate image blocks (when editing in the Wagtail interface), so it's probably not the end of the world. Just try to make the alt text as useful or unobtrusive as possible.

File: cfgov/v1/jinja2/v1/story_page.html
Comment (Scotchester) [inline cfgov/v1/jinja2/v1/story_page.html:36]: Did some testing and I am pretty confident this can be simplified to just this:

```suggestion
{% block content_main %}
    {% for block in page.content %}
        <div class="block{{ ' block__flush-top' if loop.first else '' }}">
            {% include_block block %}
        </div>
    {% endfor %}
{% endblock %}
```

(`render_block` is an older custom method that should eventually be completely replaced by the Wagtail native `include_block`)

**Review by Scotchester — CHANGES_REQUESTED** [review]
A few little bits and bobs to consider before merging. I would say that the `width-1170` one is "blocking", so that's why I chose "request changes". The others are up to you!

File: cfgov/v1/image_formats.py
Comment (contolini) [reply cfgov/v1/image_formats.py:9]: Makes sense 🎉

File: cfgov/v1/jinja2/v1/story_page.html
Comment (contolini) [reply cfgov/v1/jinja2/v1/story_page.html:36]: TY TIL. I'll pull it down and verify nothing's borked.

File: cfgov/v1/image_formats.py
Comment (contolini) [reply cfgov/v1/image_formats.py:4]: I agree with you and if we continue doing more "story" like content on cf.gov we'll want to explore more standardized methods of embedding images/diagrams/videos.

File: cfgov/v1/models/story_page.py
Comment (contolini) [reply cfgov/v1/models/story_page.py:12]: I hear what you're saying but that feels risky because I can see story pages diverging from campaign pages in the future and some cross-contamination occurring. I suppose we could just redefine `StoryHeader` when that happens but it seems cleaner to keep them separate from the get-go.

**Review by Scotchester — APPROVED** [review]

File: cfgov/v1/image_formats.py
Comment (niqjohnson) [reply cfgov/v1/image_formats.py:4]: I'm late on this, but I just wanted to note that, as of Wagtail 2.13 (which we'll be on soon), alt text is no longer required for rich text images. We'll be able to insert decorative images with a correct, blank alt attribute once the update to 2.13 is done.

**Pull request merged**
